secboot/auth_requestor.go

```diff
@@ -37,7 +38,31 @@ import (
 type systemdAuthRequestor struct {
 }
 
-func (r *systemdAuthRequestor) askPassword(sourceDevicePath, msg, credentialName string) (string, error) {
+func getAskPasswordMessage(authType sb.UserAuthType, name, path string) (string, error) {
+	var fmtMsg string
+	switch authType {
+	case sb.UserAuthTypePassphrase:
+		fmtMsg = "Enter passphrase for %[1]s (%[2]s):"
+	case sb.UserAuthTypePIN:
+		fmtMsg = "Enter PIN for %[1]s (%[2]s):"
+	case sb.UserAuthTypeRecoveryKey:
+		fmtMsg = "Enter recovery key for %[1]s (%[2]s):"
+	case sb.UserAuthTypePassphrase | sb.UserAuthTypePIN:
+		fmtMsg = "Enter passphrase or PIN for %[1]s (%[2]s):"
+	case sb.UserAuthTypePassphrase | sb.UserAuthTypeRecoveryKey:
+		fmtMsg = "Enter passphrase or recovery key for %[1]s (%[2]s):"
+	case sb.UserAuthTypePIN | sb.UserAuthTypeRecoveryKey:
+		fmtMsg = "Enter PIN or recovery key for %[1]s (%[2]s):"
+	case sb.UserAuthTypePassphrase | sb.UserAuthTypePIN | sb.UserAuthTypeRecoveryKey:
+		fmtMsg = "Enter passphrase, PIN or recovery key for %[1]s (%[2]s):"
+	default:
+		return "", errors.New("unexpected UserAuthType")
+	}
```

Comment:
This could be built instead from the set bits, but given the limited number of configurations this is fine for now with a TODO comment.

There are plans to support authentication methods, each will double the combinations.